### ISSUE TITLE
Use multi-stage builds in Dockerfile

### ISSUE THREAD

**Issue by sckott** [FredHutch/shiny-cromwell#42]
related to #39 

While investigating #39 and use of `renv` and Docker together, I was reminded of multi-stage builds. Even if we don't use `renv` I think it'd be nice to use [multi-stage builds](https://docs.docker.com/build/building/multi-stage/) to be able to iterate faster. That is, if we set it up in a way where all the system and R dependencies are setup in the first stage, then as long as we don't touch those things in a set of commits or commit, then the image will build faster. 

here's [the dockerfile](https://github.com/FredHutch/shiny-cromwell/blob/renv/Dockerfile) for the renv work with multi-stage in use

**Comment by tefirman** [comment]
This feels like a larger effort that is better suited for v2.0. Moving to the v2.0 milestone and setting as low priority for now. @sckott, that sound okay to you?

**Comment by sckott** [comment]
Yeah, sounds good. I think in JS land dependency installation can take some time, so still a good idea to keep this in mind in case docker start times are kinda slow
